src/features/classification/applied-by-ai-classification-reason/__tests__/AppliedByAiClassificationReason.test.tsx

```diff
@@ -0,0 +1,100 @@
+    });
+
+    const renderComponent = (props = {}) => {
+        return render(<AppliedByAiClassificationReason {...defaultProps} {...props} />);
+    };
+
+    test('should render AI classification reason with icon, applied date, and reasoning', () => {
+        const expectedIconSize = '1.25rem';
+
+        renderComponent();
+
+        const boxAiIcon = screen.getByTestId('box-ai-icon');
+        const appliedByWithDate = screen.getByRole('heading', {
+            name: messages.appliedByBoxAiOnDate.defaultMessage.replace('{modifiedAt}', modifiedAtDisplayDate),
+        });
+        const reasonText = screen.getByText(defaultProps.answer);
+        const citationsLabel = screen.queryByTestId('content-answers-references-label');
+        const noReferencesIconContainer = screen.queryByTestId('content-answers-references-no-references');
+
+        expect(boxAiIcon).toBeVisible();
+        expect(boxAiIcon).toHaveAttribute('height', expectedIconSize);
+        expect(boxAiIcon).toHaveAttribute('width', expectedIconSize);
+        expect(appliedByWithDate).toBeVisible();
+        expect(reasonText).toBeVisible();
+        // Assert none of the Reference components are rendered
+        expect(citationsLabel).toBeNull();
+        expect(noReferencesIconContainer).toBeNull();
+    });
+
+    test('should render no references icon when an empty citations array provided', () => {
+        renderComponent({ citations: [] });
+
+        const noReferencesIconContainer = screen.queryByTestId('content-answers-references-no-references');
+        const noReferencesIcon = noReferencesIconContainer.querySelector('svg');
+
+        expect(noReferencesIcon).toBeVisible();
+    });
+
+    test('should render references when non-empty citations are provided', () => {
+        const expectedCitationsCount = 5;
+        const expectedCitations = Array.from({ length: expectedCitationsCount }, () => ({
+            content: 'file content for citation',
+            fileId: 'fileId',
+            location: 'cited location',
+            title: 'file title',
+        }));
+
+        renderComponent({ citations: expectedCitations });
+
+        const citationsLabel = screen.queryByTestId('content-answers-references-label');
+        const citationElements = screen.getAllByTestId('content-answers-citation-status');
+
+        expect(citationsLabel).toBeVisible();
+        expect(citationElements).toHaveLength(expectedCitationsCount);
+    });
+
+    test.each([null, undefined, 'invalid date str'])(
+        'should render applied by without date when modifiedAt is invalid: %s',
+        invalidModifiedAt => {
+            renderComponent({ modifiedAt: invalidModifiedAt });
+
+            const appliedByWithoutDate = screen.getByRole('heading', {
+                name: messages.appliedByBoxAi.defaultMessage,
+            });
+
+            expect(appliedByWithoutDate).toBeVisible();
+        },
+    );
+
+    test('renders long answer text correctly', () => {
```

Comment:
This and test below not following naming convention (i.e., `should do x when y`)